bot/exts/moderation/infraction/_utils.py

```diff
@@ -280,3 +283,23 @@ async def send_private_embed(user: MemberOrUser, embed: discord.Embed) -> bool:
             "The user either could not be retrieved or probably disabled their DMs."
         )
         return False
+
+
+class StaffBanConfirmationView(interactions.ViewWithUserAndRoleCheck):
+    """The confirmation view that is sent when a moderator attempts to ban a staff member."""
+
+    @discord.ui.button(label="Confirm", style=ButtonStyle.red)
+    async def confirm(self, interaction: Interaction, button: Button) -> None:
+        """Callback coroutine that is called when the "confirm" button is pressed."""
+        await interaction.response.defer() # `ViewWithuserAndRoleCheck` view will remove the buttons
```

Comment:
When I press confirm or cancel the buttons seem to remain on the embed, so i'm not sure that this is working. `ViewWithuserAndRoleCheck` only seems to remove them when the embed times out, but I think we also want to remove them when a button is pressed.

Reply:
Fixed by merging in `main` so the latest version of bot core is used which does implement the automatic view removal.